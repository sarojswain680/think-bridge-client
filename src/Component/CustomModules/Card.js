import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import { red } from '@material-ui/core/colors';
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import React from 'react';
import './common.css';

const useStyles = makeStyles((theme) => ({
    root: {
        maxWidth: 345,
        margin: 20,
        boxShadow: "10px 10px 5px grey"
    },
    media: {
        height: 0,
        paddingTop: '100.25%',
        backgroundColor: '#D3D3D3',
        margin: 10
    },
    expand: {
        transform: 'rotate(0deg)',
        marginLeft: 'auto',
        transition: theme.transitions.create('transform', {
            duration: theme.transitions.duration.shortest,
        }),
    },
    expandOpen: {
        transform: 'rotate(180deg)',
    },
    avatar: {
        backgroundColor: red[500],
    },
}));

export default function CardView(props) {
    const classes = useStyles();

    return (
        <Card className={classes.root} id="card" data-testid="button">
            <CardMedia
                className={classes.media}
                image={props.data["photo"]}
            />
            <CardContent>
                <Typography paragraph id="text">
                    {"Product Name:- " + props.data["name"]}
                </Typography>
                <Typography paragraph>
                    {"Description:- " + props.data["description"]}
                </Typography>
                <Typography paragraph>
                    {"Price:- " + props.data["price"]}
                </Typography>
            </CardContent>
        </Card>
    );
}