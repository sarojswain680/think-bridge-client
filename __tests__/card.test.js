import '@testing-library/jest-dom/extend-expect';
import { cleanup } from '@testing-library/react';
import Enzyme, { shallow } from 'enzyme';
import renderer from "react-test-renderer";
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import Card from '../src/Component/CustomModules/Card';

Enzyme.configure({ adapter: new Adapter() })

afterEach(cleanup)


it("renderers card without crashing", () => {

    const data = {
        photo: "https://res.cloudinary.com/dlrmn0ttw/image/upload/v1594216074/sample.jpg",
        name: "FalconSat",
        description: "jngverolgmropghmtrhl[mtrlme",
        price: "220",
    }

    const wrapper = shallow(<Card data={data} />);
    const root = wrapper.find('#text')
    expect(root.first().text()).toEqual('Product Name:- FalconSat')
})

it("matches snapshot", () => {
    const data = {
        photo: "https://res.cloudinary.com/dlrmn0ttw/image/upload/v1594216074/sample.jpg",
        name: "FalconSat",
        description: "jngverolgmropghmtrhl[mtrlme",
        price: "220",
        rocket: { rocket_type: "Merlin A" }
    }
    const tree = renderer.create(<Card data={data}></Card>).toJSON();

    expect(tree).toMatchSnapshot();
})