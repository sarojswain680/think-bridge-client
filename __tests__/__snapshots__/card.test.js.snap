// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  data-testid="button"
  id="card"
>
  <div
    className="MuiCardMedia-root makeStyles-media-2"
    style={
      Object {
        "backgroundImage": "url(\\"https://res.cloudinary.com/dlrmn0ttw/image/upload/v1594216074/sample.jpg\\")",
      }
    }
  />
  <div
    className="MuiCardContent-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph"
      id="text"
    >
      Product Name:- FalconSat
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph"
    >
      Description:- jngverolgmropghmtrhl[mtrlme
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph"
    >
      Price:- 220
    </p>
  </div>
</div>
`;
